app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -129,31 +134,37 @@ class ThreadAdapter(
         return ThreadViewHolder(
             ItemMessageBinding.inflate(LayoutInflater.from(parent.context), parent, false),
             shouldLoadDistantResources,
+            isForPrinting,
             threadAdapterCallbacks?.onContactClicked,
             threadAdapterCallbacks?.onAttachmentClicked,
         )
     }
 
     override fun onBindViewHolder(holder: ThreadViewHolder, position: Int, payloads: MutableList<Any>) = runCatchingRealm {
         with(holder.binding) {
-            val payload = payloads.firstOrNull()
-            if (payload !is NotifyType) {
+            if (payloads.isEmpty()) {
                 super.onBindViewHolder(holder, position, payloads)
-                return
+                return@runCatchingRealm
             }
-
-            val message = messages[position]
-
-            when (payload) {
-                NotifyType.TOGGLE_LIGHT_MODE -> {
-                    isThemeTheSameMap[message.uid] = !isThemeTheSameMap[message.uid]!!
-                    holder.toggleContentAndQuoteTheme(message.uid)
+            payloads.forEach { payload ->
+                if (payload !is NotifyType) {
+                    super.onBindViewHolder(holder, position, payloads)
+                    return@runCatchingRealm
```

Comment:
Wow, not sure we want to call super.onBindViewHolder multiple times, right?

Why did you need to change all of this part by the way?

Reply:
These changes are not relevant anymore. I'll revert back to the original.